### PR TITLE
Additional text for stop sending variants.

### PR DESCRIPTION
**Pull request by huitema**
Addresses Martin Duke second ballot comment (Issue #153).
Addresses Robert Wilton's Stop Sending and DOQ_NO_ERROR comments (issue #150)

### CODE REVIEW

File: draft-ietf-dprive-dnsoquic.md [hunk 1]
```diff
@@ -383,6 +390,12 @@ Note that this mechanism provides a way for secondaries to cancel a single zone
 transfer occurring on a given stream without having to close the QUIC
 connection.
 
+Servers MUST NOT continue processing a transaction if they receive a RESET_STREAM
+request from the client before the client indicates the STREAM FIN. If the
+server has not yet sent the response and indicated the STREAM FIN in the response
+stream or already sent a RESET_STREAM, the server MUST issue a RESET_STREAM to
+indicate that the transaction is abandoned.
```
Comment (saradickinson) [inline draft-ietf-dprive-dnsoquic.md:397]:
```suggestion
request from the client before the client indicates the STREAM FIN. The server MUST
issue a RESET_STREAM to indicate that the transaction is abandoned unless

* it has already done so for another reason or 
* it has already both sent the response and indicated the STREAM FIN.
```

File: draft-ietf-dprive-dnsoquic.md [hunk 2]
```diff
@@ -383,6 +390,12 @@ Note that this mechanism provides a way for secondaries to cancel a single zone
 transfer occurring on a given stream without having to close the QUIC
 connection.
 
+Servers MUST NOT continue processing a transaction if they receive a RESET_STREAM
```
Comment (saradickinson) [inline draft-ietf-dprive-dnsoquic.md:393]:
```suggestion
Servers MUST NOT continue processing a DNS transaction if they receive a RESET_STREAM
```

File: draft-ietf-dprive-dnsoquic.md [hunk 3]
```diff
@@ -367,9 +371,12 @@ See {{iana-error-codes}} for details on registering new error codes.
...
+a QUIC STOP_SENDING, and it SHOULD use the error code DOQ_REQUEST_CANCELLED.
+It MAY use a more specific error code registered according to {{iana-error-codes}}.
+The STOP_SENDING request may be sent at
 any time but will have no effect if the server response has already been
-acknowledged. The corresponding DNS transaction MUST be abandoned.
+sent, in which case the client will simply discard the incoming response.
+The corresponding DNS transaction MUST be abandoned.
 
 Servers that receive STOP_SENDING act in accordance with {{Section 3.5 of RFC9000}}.
 Servers MAY impose implementation limits on the total number or rate of request cancellations.
```
Comment (saradickinson) [inline draft-ietf-dprive-dnsoquic.md:382]:
```suggestion
Servers that receive STOP_SENDING act in accordance with {{Section 3.5 of RFC9000}}. Servers SHOULD
NOT continue processing a DNS transaction if they receive a STOP_SENDING.

Servers MAY impose implementation limits on the total number or rate of request cancellations.
```

**Comment by saradickinson** [comment]
New error code and text on RESET look good - thanks. 

We now have clear text on what the servers should do with the DNS transaction after getting a RESET, but Robert's query was to clarify what servers should so with the DNS transaction after receiving a STOP_SENDING. I was going to suggest using exactly the same text for the 2 scenarios but your comments about STOP_SENDING being indicative made me pause, so there is currently just a 'SHOULD NOT continue to process' suggestion. I'd like to align these two scenarios better if we can?

**Comment by huitema** [comment]
Yes, SHOULD NOT makes sense. The server side is not enforceable by the client. It may also not be realistic. By the time the server starts sending, the look up of cache, authoritative, etc., has already been done. Except in the zone transfer case, of course.

**Pull request merged**